projects/frontend/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -10,32 +15,164 @@ import { environment } from 'projects/frontend/src/environments/environment';
 export class DashboardComponent implements OnInit {
 
   private apiUrl: string;
-  constructor(private http: HttpClient) {
+
+  constructor(
+    private http: HttpClient,
+    private authService: AuthService,
+    private router: Router,
+    private taskService: TaskService,
+    private formBuilder: FormBuilder
+    ) {
     this.apiUrl = environment.API_URL;
+    if (this.authService.getLoggedInUser() == null) {
```

Comment:
Implement AuthGuard.
* Create AuthGuard - https://frontend.learn2build.in/angular/guard/
* Protect Routes - https://frontend.learn2build.in/angular/router-3/

Reply:
AuthGuard  is implemented for routes dashboard and profile!